packages/kbn-test/src/jest/mocks/apm_agent_mock.js

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
NIT: Should we migrate this file to TS? I guess we don't want to, so we don't need to import the original library to match the types (and add the dependency). Is that right?

Reply:
I started with TS (which ensured I have a full mock as I used a `jest.Mock<Agent>` as the type of this structure when coding it), but:

1. You can't use `module.export =`, and not 100% sure `export default` is exactly the same
2. I was forced to disable some eslint warnings because `jest` is undeclared (file is not considered as a test file)
3. I had an error while running `kbn bootstrap` because this module can't `import type { Agent } from "elastic-apm-node" for some reason` (probably related to bazel project dependencies)

so given that 

4. all other mocks in this folder are still in javascript

I assumed this was okay-ish

But I can retry using TS if we think this is preferable, @elastic/kibana-operations wdyt?

Reply:
I think your points are very valid! I'm just happy that we have them documented now in this PR discussion 🙂

Reply:
It might be okay to start with the current implementation, but in the long term, we need TypeScript to enforce the mock compatibility with an interface of the original package.

Reply:
> `jest` types are declared for all *.ts files

My bad, it's actually only with `.js` extension. It worked fine with `.ts` files.

> elastic-apm-node is not declared as a TYPES_DEPS dependency in

Thanks, that was just it.

Done in https://github.com/elastic/kibana/pull/120324/commits/df6e05d97d2b6cc5a6e194f7d8e08924787ae6fa